modules/internal/Bytes.chpl

```diff
@@ -0,0 +1,473 @@
+
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+The following documentation shows functions and methods used to
+manipulate and process Chapel bytes objects.
```

Comment:
documentation -> document

Add something like `a `bytes` object is similar to a `string` but allows arbitrary data to be stored in it. Methods on `bytes` that interpret the data as characters assume that the bytes are ASCII characters.

Reply:
It's not necessary right now, but as a documentation matter, it would be better to have some examples here before the casts section.